docs/rfcs/autoops-flag-trigger.md

```diff
@@ -0,0 +1,193 @@
+    `uuid`                  varchar(512) NOT NULL default '',
+    `disabled`              tinyint(1) NOT NULL DEFAULT '0',
+    `deleted`               tinyint(1) NOT NULL DEFAULT '0',
+    `created_at`            bigint       NOT NULL,
+    `updated_at`            bigint       NOT NULL,
+    PRIMARY KEY (`id`)
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
+```
+
+Table explanation:\
+`id`: The unique ID of the flag trigger.\
+`feature_id`: The ID of the feature. \
+`environment_namespace`: The namespace of the environment.\
+`name`: The name of the flag trigger.\
+`type`: The type of the flag trigger, it can be **webhook**, etc.\
+`action`: The action of the flag trigger, it can be turned on(1)/off(0) the feature flag.\
+`description`: The description of the flag trigger.\
+`trigger_times`: The times of the flag trigger has been triggered. \
+`last_triggered_at`: The last time of the flag trigger has been triggered.\
+`url`: The URL of the flag trigger. \
+`disabled`: The flag trigger is disabled or not.\
+`deleted`: The flag trigger is deleted or not.
+
+* Reuse the current implementation to encrypt/decrypt the URL.
+
+### AutoOps Server
+
+* Add new API to manage the flag triggers.
+* As we are supporting webhook trigger first, but in the feature we will support more trigger types, so we need to
+  design the trigger architecture to support more trigger types. But one thing is that we should make the trigger
+  architecture simple and easy to use.
+
+#### AutoOps Flag Trigger API
+
+- Create a new flag trigger
+
+```Grpc
+rpc CreateFlagTrigger(CreateFlagTriggerRequest) returns (CreateFlagTriggerResponse) {}
+```
+
+```proto
+message CreateFlagTriggerRequest {
+    string feature_id = 1;
+    string environment_namespace = 2;
+    string name = 3;
+    int32 type = 4;
+    int32 action = 5;
+    string description = 6;
+}
+```
+
+- Update a flag trigger
+
+```Grpc
+rpc UpdateFlagTrigger(UpdateFlagTriggerRequest) returns (UpdateFlagTriggerResponse) {}
+```
+
+```proto
+message UpdateFlagTriggerRequest {
+    string id = 1;
+    string feature_id = 2;
+    string environment_namespace = 3;
+    string name = 4;
+    int32 type = 5;
+    int32 action = 6;
+    string description = 7;
+}
+```
+
+- Delete a flag trigger
+
+```Grpc
+rpc DeleteFlagTrigger(DeleteFlagTriggerRequest) returns (DeleteFlagTriggerResponse) {}
+```
+
+```proto
+message DeleteFlagTriggerRequest {
+    string id = 1;
+}
+```
+
+- Get a flag trigger
+
+```Grpc
+rpc GetFlagTrigger(GetFlagTriggerRequest) returns (GetFlagTriggerResponse) {}
+```
+
+```proto
+message GetFlagTriggerRequest {
+    string id = 1;
+}
+```
+
+- List flag triggers
+
+```Grpc
+rpc ListFlagTriggers(ListFlagTriggersRequest) returns (ListFlagTriggersResponse) {}
+```
+
+```proto
+message ListFlagTriggersRequest {
+    string feature_id = 1;
+    string environment_namespace = 2;
+    int32 page = 3;
+    int32 page_size = 4;
+}
+```
+
+#### AutoOps Trigger: WebHook
```

Comment:
Due to security reasons, we shouldn't show the URL more than once to the user on the console UI.
Can you add it to the RFC how are you going to deal with this when sending the response for the `ListFlagTriggersRequest`?

Reply:
OK, I think I forget this use case.

And I have checked the `FeatBit` yesterday, they just show the url after user create the trigger for the first time. 

So I think it would be easy to achieve this:

- we will return the url in the `CreateFlagTrigger` API response
- return masked url in `ListFlagTrigger` and `GetFlagTrigger` API